### DISCUSSION TITLE
umi 3 添加国际化 部署线上之后 页面空白 控制台报错

### DISCUSSION THREAD

**Discussion by Cooper1886** [umijs/umi#12445] [Q&A]
"umi": "^3.4.8",
最近项目需要国际化 用了umi的多语言方案 
本地跑没有问题但是部署线上之后页面打开是空白的
控制台有一个报错，编译后的代码不知道怎么排查有大佬遇到过吗，求指教
<img width="648" alt="image" src="https://github.com/umijs/umi/assets/107901138/be52615b-a52d-4a9d-ba07-79fc256fad6e">
<img width="448" alt="image" src="https://github.com/umijs/umi/assets/107901138/73788576-8998-41af-ba2c-42d759f7266d">

**Comment by fz6m** [comment]
umi 3 已经不维护了，建议升级到 umi 4 。

遇到 undefined 问题不一定是国际化导致的，它可能是任意的一个地方运行 undefined 了，比如某段逻辑等，在本地复现不了的话，就先尝试本地连接线上 api 接口来调试；或者开启构建生成 source-map ，然后把 source map 发布上去，在预发布环境或正式环境里测试（如果要在正式环境测试的话，请先在 nginx 等网关对 source map 文件后缀做一定的安全限制，如内网 ip 或请求头等限制；或者在生产环境下，控制台里选择使用本地 source map 来调试，具体方法请搜索引擎搜索），这样就知道哪一行报错了。
